test/unit_tests/chess_compiler_tests/01_precompiled_core_function/kernel.cc

```diff
@@ -10,13 +10,15 @@
 
 #include "kernel.h"
 
-void func(int32_t *a, int32_t *b)
+void func(int32_t *a, int32_t *b, int64_t lock)
```

Comment:
Bad for the carbon footprint to have 64 bits here?

Reply:
Yes.. this is the main reason why this hasn't been checked in.  Today, it's relatively easy to lower from locks to a constant 'index' type in MLIR, but the size of the index type is something 'implementation defined' (by default 64 bits).   This really should be cleaner in the lowering to result in something more convenient.